torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -98,6 +98,20 @@ std::vector<at::Tensor> ProcessGroupMPI::WorkMPI::result() {
   return outputTensors_;
 }
 
+c10::intrusive_ptr<c10::ivalue::Future> ProcessGroupMPI::WorkMPI::getFuture() {
+  return future_;
+}
+
+void ProcessGroupMPI::WorkMPI::finishCompleteErrorFuture(std::exception_ptr eptr) {
+  future_->setError(eptr);
+  finish(eptr);
+}
+
+void ProcessGroupMPI::WorkMPI::finishCompleteFuture() {
+  future_->markCompleted(at::IValue(outputTensors_));
+  finish();
+}
+
 ProcessGroupMPI::AsyncWork::AsyncWork(
```

Comment:
Do we plan to cover send/recv APIs as well in this release, or do we plan to focus on collectives for now? If it's the later, let's create an issue to track.

Reply:
let's focus on collectives. send/recv work will require changing how AsyncWork is handled. although we have a good idea what we'd like it to be (given @lw comments), it could be a breaking change for users who might rely on async/wait behavior.

will file an issue.